### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by vadym-khodak**
## Changes
- add Makefile
### Python:
- decrease the number of retries from 5 to 3
- decrease time sleep after a failed attempt
- use environment variables for hosts and ports
- add a route for ready testing
- add tests
### Docker:
- fix python version for docker image
- update Dockerfile to copy only the necessary files for the python web app
- delete unnecessary volume from docker-compose
- update docker-compose to run tests
### Kubernetes:
- change a label for Redis deployment and svc
- added `readinessProbe`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,9 @@
-FROM python:3.8-alpine
+FROM python:3.9-alpine
 
 MAINTAINER Vadym Khodak
 
-ADD . /src
+COPY app.py /src/app.py
```
Comment (inesusvet) [inline Dockerfile:5]: Like!

File: app.py [hunk 2]
```diff
@@ -7,15 +7,23 @@
 from flask import Flask, session
 
 app = Flask(__name__)
-app.config.update(SECRET_KEY=os.environ.get("FLASK_SECRET_KEY", "some strong secret key"))
-cache = redis.Redis(host="redis", port=6379)
+app.config.update(SECRET_KEY=os.getenv("FLASK_SECRET_KEY", "some strong secret key"))
+cache = redis.Redis(host=os.getenv("REDIS_HOST", "redis"), port=os.getenv("REDIS_PORT", 6379))
```
Comment (inesusvet) [inline app.py:11]: Minor: 'os.getenv' returns string or None. But I think that integer is expected as redis port

File: app.py [hunk 3]
```diff
@@ -7,15 +7,23 @@
...
+
+
+@app.route("/ready")
+def ready():
+    try:
+        return str(cache.ping()), 200
```
Comment (inesusvet) [inline app.py:17]: This is great as the app depends on Redis and cannot be considered "ready" until is able to connect to it

File: app.py [hunk 4]
```diff
@@ -24,12 +32,12 @@ def index():
             if retries == 0:
                 raise err
             retries -= 1
-            time.sleep(0.5)
+            time.sleep(0.1)
```
Comment (inesusvet) [inline app.py:35]: Minor: a constant or env variable is preferable instead of hard coded value

**Review by inesusvet — APPROVED** [review]
This is a great piece of work 👍

**Comment by vadym-khodak** [comment]
@inesusvet, I updated based on your comments and added simple tests

**Comment by inesusvet** [comment]
Looks even better now 😎

**Pull request merged**